src/backend/cdb/motion/ic_udpifc.c

```diff
@@ -1219,8 +1221,7 @@ setupUDPListeningSocket(int *listenerSocketFd, uint16 *listenerPort, int *txFami
 	 * Restrict what IP address we will listen on to just the one that was
 	 * used to create this QE session.
 	 */
-	Assert(interconnect_address && strlen(interconnect_address) > 0);
-	ret = pg_getaddrinfo_all(interconnect_address, NULL, &hints, &addrs);
+	ret = pg_getaddrinfo_all(interconnect_address, service, &hints, &addrs);
```

Comment:
a bit not related directly to this delta, but anyway better to hightlight it:

`pg_getaddrinfo_all` may call `getaddrinfo_unix`. In `getaddrinfo_unix` at the very beginning of the function:
```
	struct addrinfo hints;
	struct addrinfo *aip;
	struct sockaddr_un *unp;

	*result = NULL;

	MemSet(&hints, 0, sizeof(hints));

	if (strlen(path) >= sizeof(unp->sun_path))
		return EAI_FAIL;
```
The `unp` is an automatic variable (pointer to a struct) and it is not initialized.
So, the value is undefined. And before any assignment, the code deferences this pointer in `unp->sun_path`.
Does look like a problem.